### PR TITLE
[`AdaLora`] Fix adalora inference issue

### PR DESCRIPTION
**Pull request by younesbelkada**
Fixes https://github.com/huggingface/peft/issues/571

As stated by @BenjaminBossan the regularization term is only computed in training mode, i.e. when `outputs.loss` is different than None

Can confirm this snippet that is not working on main 

```python
import torch
from peft import AdaLoraConfig, get_peft_model, AutoPeftModelForCausalLM
from transformers import AutoModelForCausalLM 

model_id = "facebook/opt-350m"
peft_config = AdaLoraConfig(
    task_type="CAUSAL_LM"
)

model = AutoModelForCausalLM.from_pretrained(model_id)
dummy_input = torch.LongTensor([[1, 2, 3, 4, 5]])

model = get_peft_model(model, peft_config)
model.save_pretrained("test-adalora")

model = AutoPeftModelForCausalLM.from_pretrained("test-adalora")
_ = model(dummy_input)
```

Now works

Also fixes an issue when creating an adalora model with dropout == 0.

### CODE REVIEW

**Review by BenjaminBossan — APPROVED** [review]
LGTM, thanks!

The `trainable_adapter_name` parameter is also used in this method:

https://github.com/huggingface/peft/blob/b15c18593975d345b3857faba4860c0d68c5d287/src/peft/tuners/adalora.py#L297

But it probably makes no sense to call it in inference mode, so I guess we can leave it as is.
